libcni/conf.go

```diff
@@ -172,7 +181,23 @@ func LoadConfList(dir, name string) (*NetworkConfigList, error) {
 			return conf, nil
 		}
 	}
-	return nil, fmt.Errorf(`no net configuration list with name "%s" in %s`, name, dir)
+
+	// Try and load a network configuration file (instead of list)
+	// from the same name, then upconvert.
+	singleConf, err := LoadConf(dir, name)
+	if err != nil {
+		switch {
+		// no files at all
+		case len(files) == 0 && err == NoConfigsFoundError:
+			return nil, err
+		// some files found
```

Comment:
could be clearer - "some files found by `ConfFiles`, but not one matching the required name" ?